### PR TITLE
Uf 3542 update status when new info

### PR DESCRIPTION
**Pull request by MattiasGradinApendo**

### CODE REVIEW

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/BusinessRuleWorker.java [hunk 1]
```diff
@@ -39,7 +40,8 @@ public void execute(ExternalTask externalTask, ExternalTaskService externalTaskS
 		String caseId = externalTask.getVariable("caseNumber");
 		Optional<ErrandDTO> errandDTOOptional = Optional.empty();
 		try {
-			ErrandDTO errand = Optional.ofNullable(getErrand(externalTask)).orElseThrow(() -> Problem.valueOf(Status.NOT_FOUND));
```
Comment (josc04) [inline src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/BusinessRuleWorker.java:42]: I think the compressed version (row 42) is nicer than the two rows (43 and 44) that does the same thing. Why the change?

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java [hunk 2]
```diff
@@ -49,13 +53,28 @@ public static PatchDecisionDTO toPatchDecision(DecisionDTO decision) {
...
 
+	public static StatusDTO toStatusUpdated(ErrandDTO errandDTO){
+		StatusDTO status = new StatusDTO().dateTime(now());
+
+		Optional<StatusDTO> isStatusNeedCompletion = errandDTO.getStatuses()
+				.stream()
+				.filter(statusDTO -> STATUS_COMPLETION_NEEDED.equals(statusDTO.getStatusType()))
+				.findAny();
+		if (isStatusNeedCompletion.isPresent()){
+			status.setStatusType(STATUS_CASE_UPDATED_COMPLETED);
```
Comment (josc04) [inline src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java:71]: Row 71 and 72 can be replaced with
```
return status.statusType(STATUS_CASE_UPDATED_COMPLETED);
```

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java [hunk 3]
```diff
@@ -49,13 +53,28 @@ public static PatchDecisionDTO toPatchDecision(DecisionDTO decision) {
...
+			return status;
+		}
+		status.setStatusType(STATUS_CASE_PROCESSED);
```
Comment (josc04) [inline src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java:74]: Same here, replace 74 and 75 with 
```
return status.statusType(STATUS_CASE_PROCESSED);
```

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java [hunk 4]
```diff
@@ -49,13 +53,28 @@ public static PatchDecisionDTO toPatchDecision(DecisionDTO decision) {
...
+		return status;
+	}
+
 	private static DecisionDTO evaluateCitizenStatus(DecisionDTO decision, boolean isCitizen) {
 		if (isCitizen) {
 			decision.decisionOutcome(DecisionOutcomeEnum.APPROVAL).description(IS_CITIZEN);
```
Comment (josc04) [inline src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java:80]: Change to following:
```
	private static DecisionDTO evaluateCitizenStatus(DecisionDTO decision, boolean isCitizen) {
		if (isCitizen) {
			return decision.decisionOutcome(DecisionOutcomeEnum.APPROVAL).description(IS_CITIZEN);
		}
		return decision.decisionOutcome(DecisionOutcomeEnum.REJECTION).description(IS_NOT_CITIZEN);
	}
```

File: src/test/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapperTest.java [hunk 5]
```diff
@@ -21,17 +26,21 @@
...
 	private static final String NOT_CITIZEN = "Personen är inte skriven i Sundsvalls kommun. Rekommenderar därför avslag på ansökan.";
 	private static final String STATUS_COMPLETION_NEEDED = "Väntar på komplettering";
+
+
 	@Mock
 	DecisionDTO decisionMock;
 
 	@ParameterizedTest
 	@ValueSource(booleans = { true, false })
+	@DisplayName("Should create a Decision bases on citizen status")
```
Comment (josc04) [inline src/test/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapperTest.java:43]: `Should create a Decision bases on citizen status` -> `Should create a Decision based on citizen status`?

File: src/test/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapperTest.java [hunk 6]
```diff
@@ -6,9 +6,14 @@
 import static org.assertj.core.api.Assertions.within;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
+import static se.sundsvall.processengine.parkingpermit.Constants.STATUS_CASE_UPDATED_COMPLETED;
+import static se.sundsvall.processengine.parkingpermit.integration.casedata.mapper.CaseDataMapper.toStatusUpdated;
 
 import java.util.stream.Stream;
 
+import generated.se.sundsvall.casedata.StatusDTO;
+import org.junit.jupiter.api.Disabled;
```
Comment (josc04) [inline src/test/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapperTest.java:15]: Remove unused import

File: src/test/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapperTest.java [hunk 7]
```diff
@@ -102,16 +114,27 @@ void toStatusFromIncompleteErrandException() {
...
 
-		final var status = CaseDataMapper.toStatus(e);
+		final var status = CaseDataMapper.toStatusCompletionNeeded(e);
 
 		assertThat(status.getStatusType()).isEqualTo(STATUS_COMPLETION_NEEDED);
 		assertThat(status.getDescription()).isEqualTo(message);
 		assertThat(status.getDateTime()).isCloseTo(now(), within(2, SECONDS));
 	}
 
+	@Test
```
Comment (josc04) [inline src/test/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapperTest.java:124]: Add test to cover when method returns a status with type STATUS_CASE_PROCESSED (that is, when no status with STATUS_COMPLETION_NEEDED is found in the errand).

File: src/test/java/se/sundsvall/processengine/parkingpermit/integration/camunda/BusinessRuleWorkerTest.java [hunk 8]
```diff
@@ -148,5 +148,6 @@ void execute_throwsIncompleteErrandException() {
 		assertThat(statusCaptor.getValue().get(0).getDescription()).isEqualTo(message);
 		assertThat(statusCaptor.getValue().get(0).getDateTime()).isCloseTo(now(), within(2, SECONDS));
 	}
+
```
Comment (josc04) [inline src/test/java/se/sundsvall/processengine/parkingpermit/integration/camunda/BusinessRuleWorkerTest.java:151]: Remove unused import

**Comment by josc04** [comment]
Also, the model for decision (beslut.bpmn) seems to have been updated, so the model test should also be changed to check the possible paths that the process can take. Right now it seems to only check the happy path...

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java [hunk 9]
```diff
@@ -49,13 +53,28 @@ public static PatchDecisionDTO toPatchDecision(DecisionDTO decision) {
 		return patchDecisionDTO;
 	}
 
-	public static StatusDTO toStatus(IncompleteErrandException e) {
+	public static StatusDTO toStatusCompletionNeeded(IncompleteErrandException e) {
 		return new StatusDTO()
 			.statusType(STATUS_COMPLETION_NEEDED)
 			.description(isNull(e) ? "Ärendet behöver kompletteras med ytterligare information för att beslut ska kunna tas." : e.getMessage())
```
[the remark this answers is not in the document]
Comment (MattiasGradinApendo) [reply src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java:59]: Yeah it should be, will fix

File: src/test/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapperTest.java (hunk 7 above)
Comment (MattiasGradinApendo) [reply src/test/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapperTest.java:122]: Test added

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java (hunk 2 above)
Comment (MattiasGradinApendo) [reply src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java]: Fixed :)

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java (hunk 3 above)
Comment (MattiasGradinApendo) [reply src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java]: Fixed

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java (hunk 4 above)
Comment (MattiasGradinApendo) [reply src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java]: Good idea, fixed

**Review by josc04 — APPROVED** [review]
